lib/bloc/fee_options/fee_options_bloc.dart

```diff
@@ -34,39 +38,57 @@ class FeeOptionsBloc extends Cubit<FeeOptionsState> {
   }
 
   Future<List<FeeOption>> _constructFeeOptionList(
-    String toAddress,
-    RecommendedFees recommendedFees,
-  ) async {
+      {required String toAddress, required RecommendedFees recommendedFees, String? swapAddress}) async {
     final List<FeeOption> feeOptions = [
       FeeOption(
         processingSpeed: ProcessingSpeed.economy,
         waitingTime: const Duration(minutes: 60),
-        fee: await _calculateTransactionFee(toAddress, recommendedFees.hourFee),
+        fee: swapAddress == null
+            ? await _calculateSweepTransactionFee(toAddress: toAddress, satPerVbyte: recommendedFees.hourFee)
+            : await _calculateRefundTransactionFee(
+                satPerVbyte: recommendedFees.hourFee, toAddress: toAddress, swapAddress: swapAddress),
```

Comment:
Let's remove the null check here and pass all values to the umbrella function 
`_calculateTransactionFee({String? swapAddress, required String toAddress, required int satPerVbyte,})` 
and decide if it's a sweep or refund if swapAddress is null or not inside that function.

Reply:
Good idea,  _calculateTransactionFee handles both cases now.